### ISSUE TITLE
Attribute access of "optional" objects should raise an error

### ISSUE THREAD

**Issue by davidatbu** [microsoft/pyright#1164]
First off, thanks again for an impressively performant type checker!

**Describe the bug**
Trying to access the attributes of items annotated `Optional` should raise an error.

**To Reproduce**
Run `pyright` on the following piece of code.
```python
from typing import Optional

class A:
    a_attr: int=4

def main(optional_a: Optional[A]) -> None:
    print(optional_a.a_attr)
```

**Expected behavior**
Pyright should complain that `optional_a` can possibly be `None`. I tried mypy, and it does indeed complain like this.

**VS Code extension or command-line**
I ran `pyright test.py` from the command line, where `test.py` contained the python code above.

**Additional context**
Pyright version 1.1.85

**Comment by erictraut** [comment]
Pyright will report this error if you enable the `reportOptionalMemberAccess` diagnostic rule. By default, it is disabled in normal type checking mode. For a full list of configuration options, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/configuration.md).

**Comment by davidatbu** [comment]
Many thanks Eric! My bad for not reading enough in the docs.
